### PR TITLE
#151948280 User should be able to configure periodic reports

### PR DESCRIPTION
**Pull request by kasulani**
#### What does this PR do?
* Adds a feature User should be able to configure periodic reports, for example monthly, weekly, daily.
#### Description of the task to be completed?
* The dashboard only allows subscribing to monthly report. This should be changed to be user configurable to either be daily, weekly or monthly.
#### How should this manually be tested?
* First run automatic tests from the project root directory and ensure they all pass; 
>python manage.py test hc.accounts.tests_profile 
* To manually test this PR; 
1. Log into the application
2. Navigate to the account settings button at the top right corner of your screen and click on it
3. Navigate to the periodic reports section and click on the configure button. This brings up a modal form where you can select the different reporting periods user can choose from
4. After selecting a reporting period click on the save button, to apply changes. See screenshots section for more details
#### Pivotal Tracker story?
* See [Feature #151948280](https://www.pivotaltracker.com/story/show/151948280)
##### Step 1:
screen shot 2017-11-02 at 12 14 40

##### Step 2:
screen shot 2017-11-02 at 12 13 27

##### Step 3:
screen shot 2017-11-02 at 12 13 54

#### Questions:

### CODE REVIEW

**Comment by ivantoz** [comment]
- your PR heading and template does not follow our guidelines
- Your commit messages does not adhere to our standards
- Your PR description should include screenshots of the feature that you developed
- You should also provide manual description of how to test the feature that you delivered

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+[![Build Status](https://travis-ci.org/andela/healthchecks-apollo.svg?branch=develop)](https://travis-ci.org/andela/healthchecks-apollo)
```
Comment (ivantoz) [inline README.md]: Why is your PR changing all this file?

File: Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: gunicorn hc.wsgi --log-file -
```
Comment (ivantoz) [inline Procfile]: Why are you changing this file?

File: hc/accounts/models.py [hunk 3]
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
     def send_report(self):
         # reset next report date first:
         now = timezone.now()
-        self.next_report_date = now + timedelta(days=30)
+        if self.report_period == 0:  # monthly report
```
Comment (ivantoz) [inline hc/accounts/models.py]: Can you use enum here for the purpose of understanding the types of report period

File: hc/accounts/models.py [hunk 4]
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
...
+            self.next_report_date = now + timedelta(days=30)
+        elif self.report_period == 1:  # weekly report
```
Comment (ivantoz) [inline hc/accounts/models.py]: same for this

File: hc/accounts/models.py [hunk 5]
```diff
@@ -56,17 +57,23 @@ def set_api_key(self):
...
+            self.next_report_date = now + timedelta(days=7)
+        elif self.report_period == 2:  # daily report
+            self.next_report_date = now + timedelta(days=1)
```
Comment (ivantoz) [inline hc/accounts/models.py]: same here

File: templates/accounts/profile.html [hunk 6]
```diff
@@ -5,323 +5,401 @@
...
+                                                    type="checkbox"
+                                                    {% if profile.reports_allowed %} checked {% endif %}>
+                                            Send me a summary of my checks
+                                        </label>
+
+                                        <h4>How often do you want to get reports?</h4><br/>
+                                        <label>
+                                            <input
+                                                    name="report_period"
+                                                    type="radio"
```
Comment (ivantoz) [inline templates/accounts/profile.html]: Can you use django forms here for the purpose of learning

**Review by ivantoz — CHANGES_REQUESTED** [review]

File: hc/accounts/models.py (hunk 3 above)
Comment (kasulani) [reply hc/accounts/models.py]: We are fixing this

File: hc/accounts/models.py (hunk 5 above)
Comment (jokamjohn) [reply hc/accounts/models.py]: Working on it

File: hc/accounts/models.py (hunk 4 above)
Comment (jokamjohn) [reply hc/accounts/models.py]: working on it

File: templates/accounts/profile.html (hunk 6 above)
Comment (jokamjohn) [reply templates/accounts/profile.html]: Working on it

**Review by ivantoz — APPROVED** [review]

**Pull request merged**